subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppCachingIntegrationTest.groovy

```diff
@@ -50,30 +53,99 @@ class CppCachingIntegrationTest extends AbstractCppInstalledToolChainIntegration
         }
     }
 
-    def 'compilation can be cached'() {
-        setupProjectInDirectory(temporaryFolder.testDirectory)
+    @Unroll
+    def 'compilation can be cached (#buildType)'() {
+        setupProject()
+        this.buildType = buildType
 
         when:
-        withBuildCache().succeeds compilationTask
+        withBuildCache().run compilationTask
 
         then:
         compileIsNotCached()
 
         when:
-        withBuildCache().succeeds 'clean', ':installDebug'
+        withBuildCache().run 'clean', installTask(buildType)
```

Comment:
Why change `succeeds` to `run`?

Reply:
After reading some other tests, I find it nicer to use `run` in the `when` block. Succeeds is probably better for the `expect`/`then` block. If you feel strongly, I can change it back to `succeeds`.